wordpress-mobile/WordPress-Android
# Issue/13329 backup restore UI cycle 1

Parent #13328 & #13329

This PR makes improvements to the shared backup download and restore screens
- Updates margins/styles for subheader, additional information and checkboxes (these are shared between features)
- Moves layout manager from RestoreFragment and BackupDownloadFragment to their xml equivalents
- Adds `CheckboxSpannableLabel` to support multi color text in the checkbox label. Having this functionality in a separate class supports the unit test needs for the caller.

Notes:
1. Text has not been finalized 
2. Restore progress view labels not been updated to show label detail (in PR #13900)

**To test:**
Ensure that all states are rendered properly

|backup download details|progress|complete|
|---|---|---|
|![details](https://user-images.githubusercontent.com/506707/106051725-d5126900-60b6-11eb-867c-bd673f14d0e0.png)|![progress](https://user-images.githubusercontent.com/506707/106051744-d9d71d00-60b6-11eb-8cec-30288ca2cd1c.png)|![complete](https://user-images.githubusercontent.com/506707/106051754-de9bd100-60b6-11eb-866f-dc30e3fddd82.png)

|restore details|warning|progress|complete|
|---|---|---|---|
|![details](https://user-images.githubusercontent.com/506707/106052208-52d67480-60b7-11eb-94f0-ec196ae022c2.png)|![warning](https://user-images.githubusercontent.com/506707/106052265-5d910980-60b7-11eb-91f9-b1de12004c1f.png)|![progress](https://user-images.githubusercontent.com/506707/106052288-65e94480-60b7-11eb-8dde-2fe0c403d2ce.png)|![complete](https://user-images.githubusercontent.com/506707/106052327-6f72ac80-60b7-11eb-90c1-e45b218a7676.png)

cc @osullivanchris - if you'd like to take a look and let me know what I messed up on :) 
PR submission checklist:
